athena-hbase/src/main/java/com/amazonaws/athena/connectors/hbase/connection/HbaseConnectionFactory.java

```diff
@@ -116,6 +126,39 @@ private HBaseConnection createConnection(String host, String masterPort, String
             config.set(nextConfig.getKey(), nextConfig.getValue());
         }
 
+        Map<String, String> configOptions = System.getenv();
+        boolean kerberosAuthEnabled = configOptions.get(KERBEROS_AUTH_ENABLED) != null && "true".equalsIgnoreCase(configOptions.get(KERBEROS_AUTH_ENABLED));
+        logger.info("Kerberos Authentication Enabled: " + kerberosAuthEnabled);
+        if (kerberosAuthEnabled) {
+            String keytabLocation = null;
+            config.set("hbase.rpc.protection", configOptions.get(HBASE_RPC_PROTECTION));
+            logger.info("hbase.rpc.protection: " + config.get("hbase.rpc.protection"));
+            String s3uri = configOptions.get(KERBEROS_CONFIG_FILES_S3_REFERENCE);
+            if (StringUtils.isNotBlank(s3uri)) {
+                try {
+                    Path tempDir = copyConfigFilesFromS3ToTempFolder(configOptions);
+                    logger.debug("tempDir: " + tempDir);
+                    keytabLocation = tempDir + File.separator + "hbase.keytab";
+                    System.setProperty("java.security.krb5.conf", tempDir + File.separator + "krb5.conf");
+                    logger.debug("keytabLocation: " + keytabLocation);
+                }
+                catch (Exception e) {
+                    logger.error("Exception while copying config files from S3 to temp folder: " + e.getMessage());
```

Comment:
generally speaking you want to either log or throw an exception;

Reply:
Hi @AbdulR3hman, updated in latest commit.